### ISSUE TITLE
Volume Keyframe Constant type not working properly

### ISSUE THREAD

**Issue by leodf1** [OpenShot/openshot-qt#5491]
**Describe the bug:**
The 'Constant' type keyframe for the Volume property does not affect the adjustment instantly, but has a two/four frame tail/fade to the desired level.

**Steps to reproduce the behavior:**
1.Set a Volume keyframe in the Properties of an audio track, to zero or a lower volume
2. Set the keyframe type to Constant
3. Carefully playback the audio track
4. Notice the audio fades down to the desired level over two frames past the keyframe, instead of setting the level exactly at the keyframe

**Expected behavior:**
Volume should be set to the desired level exactly at the keyframe point when using the 'Constant' type

**System Details:**
- OpenShot Version 3.1.1. daily 11751
- Operating System: Windows 10

**Comment by leodf1** [comment]
Keyframe does not affect the level change instantly at the keyframe point, but instead acts as a short Linear change over two frames past the point.  Makes it hard to remove noises etc from the audio track without precise control over the Volume keyframes.

[480P 15FPS PROFILE USED. So might happen across four frames on a 30fps profile, I don't know]

Not sure if this only affects the Volume Property or other Property keyframes too.

Had not noticed this in earlier version, so bug may have been introduced in this version of the software.

tldr:  Constant type keyframe acts as a Linear type keyframe

**Comment by Colorjet3** [comment]
Hello @leodf1.
I haven't seen a reply on this.  I tried to replicate your issue without success.

Here are the steps I did to try and replicate the issues:
1.  Imported a .mp4 file that has audio from the beginning and dragged it unto Track 5 position at 0:00.
2.  Positioned the playhead with the left edge of the clip.  This is important so you don't get animation.
3.  Right Click | Properties | Volume attribute and changed it from 1.00 to 0.00.
4.  Previewed the clip and no volume for the entire clip.
5.  Re-positioned the playhead at the left edge of the clip.
6.  Modified the Volume from 0.00 to 0.5.
7.  Previewed the clip and the volume is lower now for the entire clip.

If you are doing something different then be very clear in your step-by-step instructions so we can follow it precisely.

**Comment by dsmurl** [comment]
I'm getting that the volume property doesn't appy at all.  When I go back and click on the clip again, the volume is 1 always no matter what it was before.  In playback the property doesn't effect the volume at all either.

**Comment by Colorjet3** [comment]
Hello @dsmurl 
1.  What operating system/version are you running (windows 11?  Ubuntu 24.04? etc?
2.  Which version of OpenShot are you running (help | about OpenShot)?
3.  Is this a video/audio file or just audio?  Also, what type of a file is it (.mp3, .mp4, .mov, etc?)
4.  Please provide the exact steps you are performing (don't leave any details, including where the playhead is positioned, etc.) so we can try and replicate your issue.
